app/assets/templates/trackview.html.erb

```diff
@@ -1189,10 +1157,10 @@
 						drawPattern(context, trackset.currentPattern, displaySettings);
 					}
 
-					preview = trackset.SelectPattern({trackIndex: trackIndex, tickstart: mouseX * displaySettings.TPP});
+					var pattern = trackset.SelectPattern({trackIndex: mouse.trackIndex, _start: mouse.x * displaySettings.TPP});
 
-					if(typeof preview !== 'undefined'){
-            var patternEvent = new CustomEvent('denoto-editpattern', {"detail": rhomb.getSong().getPatterns()[preview.ID]});
+					if(typeof pattern !== 'undefined'){
+            			var patternEvent = new CustomEvent('denoto-editpattern', {"detail": rhomb.getSong().getPatterns()[pattern.ID]});

```

Comment:
There is some apparent confusion between playlist items and patterns, as demonstrated here. You're attempting to index into the pattern map using a playlist item ID (which should be '_id' and not 'ID', of course.